lib/csv.rb

```diff
@@ -1187,7 +1221,8 @@ def instance(data = $stdout, **options)
     #   See {Options for Parsing}[#class-CSV-label-Options+for+Parsing].
     def filter(input=nil, output=nil, **options)
       # parse options for input, output, or both
-      in_options, out_options = Hash.new, {row_sep: $INPUT_RECORD_SEPARATOR}
+      row_sep = (defined?(Ractor) && (Ractor.current != Ractor.main)) ? "\n" : $INPUT_RECORD_SEPARATOR
+      in_options, out_options = Hash.new, {row_sep: row_sep}
```

Comment:
Why do we need to avoid using `$INPUT_RECORD_SEPARATOR`?
It's sharable in Ractors because it's a frozen string.

Reply:
I agree that it is a frozen string, but according to the specifications $INPUT_RECORD_SEPARATOR can't be used outside the main Ractor. Unless it's outdated, I think that this was mentioned here (as $/):   https://bugs.ruby-lang.org/issues/17268. I hope I've understood correctly!

Reply:
Ah, https://bugs.ruby-lang.org/issues/18129 ?

Reply:
`$/` is deprecated since Ruby 3.0: https://bugs.ruby-lang.org/issues/14240
We may be able to replace `$INPUT_RECORD_SEPARATOR` by '"\n"' with Ruby 3.0 or later.

Reply:
Done on master.

Reply:
Could you rebase on master?
You need to resolve conflict. Sorry.